src/app/vc-receipt-status/middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response } from "express";
+import { NextFunction, RequestHandler } from "express-serve-static-core";
+import { appVcReceived } from "../../services/coreBackService";
+import { isJourneyResponse } from "../validators/postJourneyEventResponse";
+import { isAxiosError } from "axios";
+import config from "../../config/config";
+
+enum AppVcReceiptStatus {
+  COMPLETED = "COMPLETED",
+  ERROR = "ERROR",
+  PROCESSING = "PROCESSING",
+  INTERVENTION = "INTERVENTION",
+}
+
+export const getAppVcReceiptStatus: RequestHandler = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    // For browser tests
+    if (config.ENABLE_PREVIEW) {
+      res.status(200).json({ status: AppVcReceiptStatus.COMPLETED });
+      return;
+    }
+
+    const appVcResponse = await appVcReceived(req);
+
+    if (isJourneyResponse(appVcResponse.data)) {
+      req.session.journey = appVcResponse.data.journey;
+    }
```

Comment:
I think this should be made stricter, if there is no journey response the eventual continue button won't work.

It _should_ return a journey response for a successful response, so I think it should fail if not:

```suggestion
    if (!isJourneyResponse(appVcResponse.data)) {
      throw new Exception("Journey response expected from successful check app vc receipt response.");
    }
    
    req.session.journey = appVcResponse.data.journey;
```